### PR TITLE
Slap my hand

### PR DESCRIPTION
**Pull request by SomeguyManperson**
slapmyhand

DEMONSTRATION

https://streamable.com/t07aw

:cl:  Theos
rscadd: Adds high-fiving 
/:cl:

### CODE REVIEW

**Comment by jcatjmeow** [comment]
Add a video of you testing it and I'll give you a big +1 and be very happy

**Comment by E231kei500bandai** [comment]
[Supplementary Video](https://www.youtube.com/watch?v=xaeFLz0Eu6E)

File: yogstation/code/modules/mob/living/emote.dm [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	restraint_check = TRUE
+
+/datum/emote/living/raisehand/run_emote(mob/user, params)
+	. = ..()
+	var/obj/item/highfive/N = new(user)
+	if(user.put_in_hands(N))
+		to_chat(user, "<span class='notice'>You raise your hand for a high-five.</span>")
+	else
+		qdel(N)
+		to_chat(user, "<span class='warning'>You don't have any free hands high-five with.</span>")
```
Comment (Altoids1) [inline yogstation/code/modules/mob/living/emote.dm]: spellfix: You don't have any free hands **to** high-five with.

Comment (SomeguyManperson) [reply yogstation/code/modules/mob/living/emote.dm]: aaaaaa

**Comment by Nickvr628** [comment]
First handholding and now this? We are gonna have bad-deathclaw style interactions in no time!

**Comment by SomeguyManperson** [comment]
handholding is for catgirls, high-fives are for chads

**Comment by Altoids1** [comment]
The Conga PR is next

File: yogstation/code/game/objects/items/weaponry.dm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	item_flags = DROPDEL | ABSTRACT
+	attack_verb = list("is left hanging by")
+
+/mob/proc/thisisnotatrick()
+	var/obj/item/highfive/H = locate() in src
+	if(!get_active_held_item())
+		return FALSE
+	if(get_active_held_item() == H)
+		return TRUE
+	return FALSE
```
Comment (monster860) [inline yogstation/code/game/objects/items/weaponry.dm]: This code makes me want to vomit.

```dm
/mob/proc/thisisnotatrick()
	return istype(get_active_held_item(), /obj/item/highfive)
```

**Review by monster860 — CHANGES_REQUESTED** [review]

File: yogstation/code/game/objects/items/weaponry.dm (hunk 2 above)
Comment (SomeguyManperson) [reply yogstation/code/game/objects/items/weaponry.dm]: replaced the proc with istype

**Review by monster860 — APPROVED** [review]

**Pull request merged**